test/plots/barPlotTests.ts

```diff
@@ -1 +1,783 @@
+
+            const basePosition = TestMethods.numAttr(bar, basePositionAttr);
+            assert.closeTo(basePosition, baseScale.scale(datum.base) - 0.5 * baseSize, window.Pixel_CloseTo_Requirement,
+              `bar ${basePositionAttr} is correct (index ${index})`);
+
+            const valuePosition = TestMethods.numAttr(bar, valuePositionAttr);
+            const isShifted = isVertical ? (datum.value > barPlot.baselineValue()) : (datum.value < barPlot.baselineValue());
+            const expectedValuePosition = (isShifted) ? scaledBaselineValue - valueSize : scaledBaselineValue;
+            assert.closeTo(valuePosition, expectedValuePosition, window.Pixel_CloseTo_Requirement,
+              `bar ${valuePositionAttr} is correct (index ${index})`);
+          });
+        }
+
+        it("renders with no data", () => {
+          assert.doesNotThrow(() => barPlot.renderTo(svg), Error);
+          assert.strictEqual(barPlot.width(), TestMethods.numAttr(svg, "width"), "was allocated width");
+          assert.strictEqual(barPlot.height(), TestMethods.numAttr(svg, "height"), "was allocated height");
+        });
+
+        it("draws bars and baseline in correct positions", () => {
+          barPlot.addDataset(dataset);
+          barPlot.renderTo(svg);
+
+          assertCorrectRendering();
+        });
+
+        it("rerenders correctly when the baseline value is changed", () => {
+          barPlot.addDataset(dataset);
+          barPlot.renderTo(svg);
+
+          barPlot.baselineValue(1);
+          assertCorrectRendering();
+        });
+
+        it("can autorange value scale based on visible points on base scale", () => {
+          const firstTwoBaseValues = [data[0].base, data[1].base ];
+          valueScale.padProportion(0);
+          baseScale.domain(firstTwoBaseValues);
+          barPlot.addDataset(dataset);
+          barPlot.autorangeMode(valuePositionAttr);
+          barPlot.renderTo(svg);
+
+          const valueScaleDomain = valueScale.domain();
+          const expectedValueDomainMin = Math.min(data[0].value, data[1].value);
+          const expectedValueDomainMax = Math.max(data[0].value, data[1].value);
+          assert.strictEqual(valueScaleDomain[0], expectedValueDomainMin, "lower bound of domain set based on visible points");
+          assert.strictEqual(valueScaleDomain[1], expectedValueDomainMax, "upper bound of domain set based on visible points");
+        });
+
+        it("doesn't show values from outside the base scale's domain", () => {
+          baseScale.domain(["-A"]);
+          barPlot.addDataset(dataset);
+          barPlot.renderTo(svg);
+
+          assert.strictEqual(barPlot.content().selectAll("rect").size(), 0, "draws no bars when the domain contains no data points");
+        });
+
+        afterEach(function() {
+          if (this.currentTest.state === "passed") {
+            barPlot.destroy();
+            svg.remove();
+          }
+        });
```

Comment:
Done. Not sure why this comment thread appears in a weird spot on the main PR page.